shop/src/main/java/ro/msg/learning/shop/service/ProductService.java

```diff
@@ -0,0 +1,86 @@
+package ro.msg.learning.shop.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.domain.Product;
+import ro.msg.learning.shop.domain.ProductCategory;
+import ro.msg.learning.shop.dto.ProductGetDto;
+import ro.msg.learning.shop.mapper.ProductCategoryMapper;
+import ro.msg.learning.shop.mapper.ProductMapper;
+import ro.msg.learning.shop.repository.ProductRepository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class ProductService {
+    public static final String ID_INVALID = "ID invalid.";
+
+    @Autowired
+    private ProductRepository productRepository;
+
+    @Autowired
+    private ProductCategoryService productCategoryService;
+
+    @Autowired
+    private ProductCategoryMapper productCategoryMapper;
+
+    @Autowired
+    private ProductMapper productMapper;
+
+    public Product createProduct(ProductGetDto product) {
+
+        ProductCategory productCategoryFound = productCategoryService.findByName(product.getCategoryName()).orElse(null);
+
+        if (productCategoryFound == null) {
```

Comment:
find by name rturns an optional, we can check if optional is present instead of giving null and then checking again

Reply:
done.  giving null and then checking again was pointless

Reply:
i used Optional in every mehtod, i think i missed this one :)